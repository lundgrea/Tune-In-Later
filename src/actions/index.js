export const addAlbums = albums => ({
  type: "ADD_ALBUMS",
  albums
});

export const login = user => ({
  type: "LOGIN",
  user
});

export const logout = () => ({
  type: "LOGOUT"
});

export const toggleFavorite = (id) => ({
  type: "TOGGLE_FAVORITE",
  id
});

export const createUser = user => ({
  type: "CREATE_USER",
  user
});

export const handleErrors = error => ({
  type: "HANDLE_ERROR",
  error
});

export const storeFavorites = albums => ({
  type: "STORE_FAVORITES",
  albums
});

// export const deleteFavorite = albumId => ({
//   type: "DELETE_FAVORITE",
//   albumId
// });
