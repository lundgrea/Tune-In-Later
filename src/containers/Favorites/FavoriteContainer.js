import React from 'react';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import { toggleFavorite } from '../../actions';


const FavoriteContainer = ({toggleFavorite, user, favorites, dataType }) => {
  console.log('favorites', favorites)
  let filters = [];
  favorites.forEach(genre => {
    if(!filters.includes(genre.primary_genre_name)){
      filters.push(genre.primary_genre_name)
    }
  })
  let genreNodes = filters.map((genre) => {
  return (
    <>
      <label for={genre}>{genre}</label>
      <input type="radio" name="genre" value={genre} placeholder="radio buttons"></input>
    </>)
  })
  console.log('filter', filters)
    let cards = favorites.map(favorite => 
      {
      return <Card
        {...favorite}
        toggleFavorite={toggleFavorite}
        user={user}
        key={favorite.id}
        isFavorite={true}
      />
    });
 

  return (
    <section>
      <div>
        <fieldset>
          <h3>Filter By Genre</h3>
          {genreNodes}
        </fieldset>
      </div>
      {cards}
    </section>
  )
}

const mapStateToProps = store => ({
  user: store.user,
  favorites: store.favorites
})


const mapDispatchToProps = dispatch => ({
  toggleFavorite: (id) => dispatch(toggleFavorite(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(FavoriteContainer)
