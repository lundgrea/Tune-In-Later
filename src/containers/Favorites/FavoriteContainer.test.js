import React from 'react';
import { shallow } from 'enzyme';
import { FavoriteContainer, mapStateToProps, mapDispatchToProps } from './FavoriteContainer';
import { toggleFavorite } from '../../actions'

describe('Favorite Container ', () => {
  let wrapper;
  const mockUser = {
    id: 1,
    name: 'aidan',
    email: 'aidanmckay2000@gmail.com'
  }
  const mockFavorites = [
    {album_name: 'mirrorland', album_id: 12345678, primary_genre_name: 'rap'}
  ]
  const mockGenerateAlbum = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <FavoriteContainer
        user={mockUser}
        favorites={mockFavorites}
        toggleFavorite={jest.fn()}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should filter favorites based on value of filter and add album to state', () => {
    const mockEvent = { target: { value:'rap'} }
    expect(wrapper.state('filteredAlbums')).toEqual([])
    wrapper.find('input').at(1).simulate('click', mockEvent)
    expect(wrapper.state('filteredAlbums')).toEqual(mockFavorites)
  });

  it('should gather filters based on favorite albums', () => {
    wrapper.instance().gatherFilters()
    expect(wrapper.state('filters')).toEqual(['rap'])
  });

  it('should reset state on click', () => {
    wrapper.find('input').at(0).simulate('click')

    expect(wrapper.state('filteredAlbums')).toEqual([])
    expect(wrapper.state('filterButton')).toEqual('')
  })
  describe('mapStateToProps', () => {
    it('should return an object with a user object', () => {
      const mockState = {
        user: mockUser
      };
      const expected = {
        user: mockUser
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('should return an array of favorite album objects', () => {
      const mockState = {
        favorites: mockFavorites
      }
      const expected = {
        favorites: mockFavorites
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with an toggleFavorite action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleFavorite(1);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.toggleFavorite(1);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});