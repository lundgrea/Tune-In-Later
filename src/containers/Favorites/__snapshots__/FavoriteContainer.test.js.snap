// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Container  should match snapshot 1`] = `
<section>
  <div>
    <fieldset>
      <h3>
        Filter By Genre
      </h3>
      <label
        htmlFor="show all"
      >
        Show All
      </label>
      <input
        name="genre"
        onClick={[Function]}
        placeholder="radio buttons"
        type="radio"
        value="show all"
      />
      <label
        htmlFor="rap"
      >
        rap
      </label>
      <input
        name="genre"
        onClick={[Function]}
        placeholder="radio buttons"
        type="radio"
        value="rap"
      />
    </fieldset>
  </div>
  <div
    className="favorites-container"
  >
    <Connect(Card)
      album_id={12345678}
      album_name="mirrorland"
      isFavorite={true}
      primary_genre_name="rap"
      toggleFavorite={[MockFunction]}
      user={
        Object {
          "email": "aidanmckay2000@gmail.com",
          "id": 1,
          "name": "aidan",
        }
      }
    />
  </div>
</section>
`;
